src/node/internal/internal_net.ts

```diff
@@ -1753,3 +1749,529 @@ export class SocketAddress implements _SocketAddress {
+      if (rule.check(checkAddr, checkFamily)) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  get rules(): string[] {
+    return this.#rules.map((rule) => rule.toString());
+  }
+
+  [kInspect](depth: number, options: InspectOptions): string {
+    if (depth < 0) return '[BlockList]';
+
+    const opts: InspectOptions = {
+      ...options,
+      depth: options.depth == null ? null : options.depth - 1,
+    };
+
+    // @ts-expect-error TS2769 not all the overloads are compatible
+    return `BlockList {\n  rules: ${inspect(this.rules, opts)}\n}`;
+  }
+
+  toJSON(): string[] {
+    return this.rules;
+  }
+
+  fromJSON(data: string | string[]): void {
+    let rules: string[];
+
+    if (typeof data === 'string') {
+      try {
+        rules = JSON.parse(data) as string[];
+      } catch {
+        throw new ERR_INVALID_ARG_VALUE('data', data, 'must be valid JSON');
+      }
+    } else {
+      rules = data;
+    }
+
+    if (!Array.isArray(rules)) {
+      throw new ERR_INVALID_ARG_VALUE('data', data, 'must be an array');
+    }
+
+    this.#parseIPInfo(rules);
+  }
+
+  #parseIPInfo(data: string[]): void {
+    for (const item of data) {
+      if (item.includes('IPv4')) {
+        // IPv4 subnet pattern
+        const subnetMatch = item.match(
+          /Subnet: IPv4 (\d{1,3}(?:\.\d{1,3}){3})\/(\d{1,2})/
+        );
+        if (subnetMatch) {
+          const [, network, prefix] = subnetMatch;
+          this.addSubnet(network!, parseInt(prefix!, 10));
+          continue;
+        }
+
+        // IPv4 address pattern
+        const addressMatch = item.match(
+          /Address: IPv4 (\d{1,3}(?:\.\d{1,3}){3})/
+        );
+
+        if (addressMatch) {
+          const [, address] = addressMatch;
+          this.addAddress(address!);
+          continue;
+        }
+
+        // IPv4 range pattern
+        const rangeMatch = item.match(
+          /Range: IPv4 (\d{1,3}(?:\.\d{1,3}){3})-(\d{1,3}(?:\.\d{1,3}){3})/
+        );
+        if (rangeMatch) {
+          const [, start, end] = rangeMatch;
+          this.addRange(start!, end!);
+          continue;
+        }
+      }
+
+      if (item.includes('IPv6')) {
+        // IPv6 subnet pattern
+        const ipv6SubnetMatch = item.match(
+          /Subnet: IPv6 ([0-9a-fA-F:]{1,39})\/(\d{1,3})/i
+        );
+        if (ipv6SubnetMatch) {
+          const [, network, prefix] = ipv6SubnetMatch;
+          this.addSubnet(network!, parseInt(prefix!, 10), 'ipv6');
+          continue;
+        }
+
+        // IPv6 address pattern
+        const ipv6AddressMatch = item.match(
+          /Address: IPv6 ([0-9a-fA-F:]{1,39})/i
+        );
+        if (ipv6AddressMatch) {
+          const [, address] = ipv6AddressMatch;
+          this.addAddress(address!, 'ipv6');
+          continue;
+        }
+
+        // IPv6 range pattern
+        const ipv6RangeMatch = item.match(
+          /Range: IPv6 ([0-9a-fA-F:]{1,39})-([0-9a-fA-F:]{1,39})/i
```

Comment:
Can you move all of these regex instances outside of the function, in order to be reused?